### PR TITLE
4676 mixed labels in r

### PR DESCRIPTION
**Pull request by oscardssmith**
Note this was branched off of #2301 so that we could test with newer stata files, and as such should not be merged before it.

## Related Issues

- connects to #4676 : Dataverse ingest error for variables with mix of labelled and unlabelled values

## Pull Request Checklist

- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: https://github.com/IQSS/dataverse/tree/develop/scripts/database/upgrades
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/datavariable/DataVariable.java [hunk 1]
```diff
@@ -414,13 +426,21 @@ public void setUnf(String unf) {
     public Collection<VariableCategory> getCategories() {
         return this.categories;
     }
+
+    public boolean isLabled () {
```
Comment (oscardssmith) [inline src/main/java/edu/harvard/iq/dataverse/datavariable/DataVariable.java]: I'm not sure if I should remove `isLabled` the theory behind it is that it represents the fact that the SPSS and Stata way of looking at these is different from that of R, but on the other hand, it adds code that currently doesn't really have a consumer (as UNF ignores it currently, and we don't ask if something is categorical anywhere else)

File: src/main/java/edu/harvard/iq/dataverse/dataaccess/DataConverter.java [hunk 2]
```diff
@@ -220,28 +218,50 @@ private static File runFormatConversion (DataFile file, File tabFile, String for
...
+                } else if (origFormat.contains("sav")){
+                    origFormat = "sav";
+                } else if (origFormat.contains("por")){
+                    origFormat = "por";
+                }
+                    
+                try {
+                    StorageIO<DataFile> storageIO = file.getStorageIO();
+                    long size = storageIO.getAuxObjectSize("orig");
+                    File origFile = downloadFromByteChannel((ReadableByteChannel) storageIO.openAuxChannel("orig"), size);
```
Comment (oscardssmith) [inline src/main/java/edu/harvard/iq/dataverse/dataaccess/DataConverter.java]: I don't know whether this cast to `ReadableByteChannel` should instead be a change to `StorageIO`, as all current ones return `ReadableByteChannel`s. I figured that that is a bigger change, and probably needs an OK from @landreev and/or @scolapasta  first.

**Comment by pdurbin** [comment]
@oscardssmith this branch is 102 commits behind "develop":

screen shot 2018-07-09 at 10 18 24 am

Are you waiting for pull request #4708 to get merged?

**Comment by oscardssmith** [comment]
yes. Since this issue is branched off of #4708 the diff size will be about 2x as big until that gets merged.

**Comment by oscardssmith** [comment]
Note when Code Reviewing this, (at least until #2301 is merged) you should compare the diff to #2301 so you aren't looking at the changes introduced there.

**Comment by pdurbin** [comment]
> you should compare the diff to #2301

Here's a handy link: https://github.com/IQSS/dataverse/compare/2301-stata...4676-mixed-labels-in-R

**Pull request merged**